tfhe/src/integer/server_key/radix_parallel/tests_unsigned/test_slice.rs

```diff
@@ -0,0 +1,443 @@
+use std::ops::{Range, RangeBounds};
+use std::sync::Arc;
+
+use rand::prelude::*;
+
+use crate::error::InvalidRangeError;
+use crate::integer::keycache::KEY_CACHE;
+use crate::integer::server_key::radix::slice::parse_bounds;
+use crate::integer::tests::create_parametrized_test;
+use crate::integer::{IntegerKeyKind, RadixCiphertext, RadixClientKey, ServerKey};
+use crate::prelude::CastFrom;
+#[cfg(tarpaulin)]
+use crate::shortint::parameters::coverage_parameters::*;
+use crate::shortint::parameters::*;
+
+use super::{nb_tests_for_params, CpuFunctionExecutor, FunctionExecutor, PBSParameters, NB_CTXT};
+
+create_parametrized_test!(integer_unchecked_scalar_slice);
+create_parametrized_test!(integer_unchecked_scalar_slice_assign);
+create_parametrized_test!(integer_default_scalar_slice);
+create_parametrized_test!(integer_default_scalar_slice_assign);
+create_parametrized_test!(integer_smart_scalar_slice);
+create_parametrized_test!(integer_smart_scalar_slice_assign);
+
+// Reference implementation of the slice using a conversion into a string of 0/1 to do the slicing.
+fn slice_reference_impl<T, B>(value: u64, bounds: B, modulus: u64) -> u64
+where
+    B: RangeBounds<T>,
+    T: CastFrom<usize> + Copy,
+    usize: CastFrom<T>,
+{
+    let (start, end) = parse_bounds(&bounds, modulus as usize).unwrap();
+
+    let bin: String = format!("{value:064b}").chars().rev().collect();
+
+    let out_bin: String = bin[start..end].chars().rev().collect();
+    u64::from_str_radix(&out_bin, 2).unwrap_or_default()
```

Comment:
We can to this without intermediate strings: 

```rust
    // This removes the bits >= end
    let result = value << (u64::BITS - end);
    // This removes the bits < start
    result >> (start + (u64::BITS - end))
```

Reply:
This does not work if `start == end`, which I think should return 0 as "the empty slice".

But I think the string implem is closer to the definition which is why it is used in tests. But I can switch to an arithmetic implementation if you think it's better.

Reply:
Yeah for start == end a `if` would be needed

The string version is ok for tests as its not a performance sensitive thing